solver/src/main/java/org/chocosolver/solver/search/strategy/strategy/AbstractStrategy.java

```diff
@@ -39,6 +39,12 @@ public boolean init(){
         return true;
     }
 
+    /**
+     * Indicates if this strategy is learn-based may, for instance, require to be coupled with a restart strategy.
+     * @return <i>true</i> if this strategy is dynamic.
+     */
+    public abstract boolean isDynamic();
```

Comment:
I find the concept of dynamic search quite vague and possibly confusing. 
If it is really about restarts i would suggest a more explicit name, like isImprovedByRestarts. 
However, i am not 100% convinced by the idea of this feature, because it often depends on the problem.

Reply:
Indeed, the choice of the term and the concept it encapsulates are debatable. 
However, it happens quite often (and this is notably the case of the default strategy 🤦) that the strategy/restarts couple is unsuitable. 

However, this can be the subject of another RP with a more in-depth discussion.